onnxruntime/core/providers/cuda/tensor/sequence_op.h

```diff
@@ -29,7 +19,16 @@ class SequenceAt final : public CudaKernel {
     const Tensor* I = context->Input<Tensor>(1);
     ORT_ENFORCE(I != nullptr, "SequenceAt GPU: Got nullptr input for index tensor.");
 
-    int64_t idx = I->GetElementType() == ONNX_NAMESPACE::TensorProto_DataType_INT32 ? ReadIndex<int32_t>(*I, "int32_t") : ReadIndex<int64_t>(*I, "int64_t");
+    ORT_ENFORCE(I->IsDataType<int32_t>() || I->IsDataType<int64_t>(),
+                "Indices need to be of types int32 or int64");
+
```

Comment:
Waste of bytes. The kernel type constraints enforce this. 

Same goes for checking inputs aren't null. That could only occur if we had nothing producing/providing the input, which should have been detected during Graph resolve

Reply:
Removed the checking for the indices type. Also removed the checks for inputs and outputs. FWIW: They were present even before the change.